pgrx/src/datum/array.rs

```diff
@@ -204,98 +227,28 @@ impl<'a, T: FromDatum> Array<'a, T> {
-                // already, and I'd rather err on the side of correctness.
-                //
-                // Size::Fixed(size) if size as usize == std::mem::size_of::<T>() => unsafe {
-                //     // short-circuit if the size of the element matches the size of `T`.
-                //     // This most likely means that the element Datum actually represents the same
-                //     // type as the rust `T`
-                //
-                //     Some(ptr.cast::<T>().read())
-                // },
-                Size::Fixed(size) => {
-                    // copy off `size` bytes from the head of `ptr` and convert that into a `usize`
-                    // using proper platform endianness, converting it into a `Datum`
-                    #[inline(always)]
-                    fn bytes_to_datum(ptr: *const u8, size: usize) -> Datum {
-                        const USIZE_BYTE_LEN: usize = std::mem::size_of::<usize>();
-
-                        // a zero-padded buffer in which we'll store bytes so we can
-                        // ultimately make a `usize` that we convert into a `Datum`
-                        let mut buf = [0u8; USIZE_BYTE_LEN];
-
-                        match size {
-                            1..=USIZE_BYTE_LEN => unsafe {
-                                // copy to the end
-                                #[cfg(target_endian = "big")]
-                                let dst = (&mut buff[8 - size as usize..]).as_mut_ptr();
-
-                                // copy to the head
-                                #[cfg(target_endian = "little")]
-                                let dst = (&mut buf[0..]).as_mut_ptr();
-
-                                std::ptr::copy_nonoverlapping(ptr, dst, size as usize);
-                            },
-                            other => {
-                                panic!("unexpected fixed size array element size: {}", other)
-                            }
-                        }
-
-                        Datum::from(usize::from_ne_bytes(buf))
-                    }
-
-                    let datum = bytes_to_datum(ptr, size as usize);
-                    unsafe { T::from_polymorphic_datum(datum, false, self.raw.oid()) }
-                }
-
-                other => {
-                    panic!("unrecognized pass-by-value array element layout size: {:?}", other)
-                }
-            },
-            PassBy::Ref => {
-                let datum = pg_sys::Datum::from(ptr);
-                #[cfg(debug_assertions)]
-                assert_eq!(
-                    Some(datum),
-                    self._datum_slice.get().and_then(|s| unsafe { s.get(_index) }).copied()
-                );
-                unsafe { T::from_polymorphic_datum(datum, false, self.raw.oid()) }
-            }
+    unsafe fn bring_it_back_now(&self, ptr: *const u8, is_null: bool) -> Option<T> {
+        match is_null {
```

Comment:
match on bool is a bit goofy but I suppose it's fine.

Reply:
It made the code pretty.